types/lazy/lazy.go

```diff
@@ -26,6 +43,7 @@ func (z *SyncValue[T]) Set(val T) bool {
 	var wasSet bool
 	z.once.Do(func() {
 		z.v = val
+		z.err.Store(nilErrPtr)
```

Comment:
Comment that this must be after the assignment to `z.v`, otherwise this is racy?

Reply:
done, in several spots